<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>uGLWindowClass</class>
 <widget class="QMainWindow" name="uGLWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1299</width>
    <height>792</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>uGLWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="uGLSceneWidget" name="openGLWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1299</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport_as">
     <property name="title">
      <string>Export as</string>
     </property>
     <addaction name="actionExport_as_PNG"/>
    </widget>
    <addaction name="menuExport_as"/>
    <addaction name="actionImport_model"/>
   </widget>
   <widget class="QMenu" name="menuCamera">
    <property name="title">
     <string>Camera</string>
    </property>
    <addaction name="actionCamera_Translation"/>
    <addaction name="actionCameraDepth"/>
    <addaction name="actionCamera_Rotation"/>
    <addaction name="actionCamera_Roll"/>
    <addaction name="actionVertigo"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuObject">
    <property name="title">
     <string>Object</string>
    </property>
    <widget class="QMenu" name="menuNew">
     <property name="title">
      <string>New</string>
     </property>
     <widget class="QMenu" name="menu3D">
      <property name="title">
       <string>3D</string>
      </property>
      <addaction name="actionCube"/>
      <addaction name="actionPyramid"/>
      <addaction name="actionBezier_surface"/>
     </widget>
     <widget class="QMenu" name="menu2D">
      <property name="title">
       <string>2D</string>
      </property>
      <addaction name="actionSquare"/>
      <addaction name="actionCircle"/>
      <addaction name="actionTriangle"/>
      <addaction name="actionPolygon"/>
      <addaction name="actionBezier_curve"/>
     </widget>
     <widget class="QMenu" name="menuTerrain">
      <property name="title">
       <string>Terrain</string>
      </property>
      <addaction name="actionGrass"/>
      <addaction name="actionWater"/>
      <addaction name="action3D_rock"/>
     </widget>
     <addaction name="menu3D"/>
     <addaction name="menu2D"/>
     <addaction name="menuTerrain"/>
    </widget>
    <addaction name="menuNew"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCamera"/>
   <addaction name="menuObject"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="locale">
    <locale language="English" country="Canada"/>
   </property>
   <property name="movable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionImport_model"/>
   <addaction name="separator"/>
   <addaction name="actionCamera_Translation"/>
   <addaction name="actionCameraDepth"/>
   <addaction name="actionCamera_Rotation"/>
   <addaction name="actionCamera_Roll"/>
   <addaction name="actionVertigo"/>
   <addaction name="separator"/>
   <addaction name="actionExport_as_PNG"/>
   <addaction name="actionModify_Skybox"/>
  </widget>
  <widget class="QDockWidget" name="shape_dockWidget">
   <property name="minimumSize">
    <size>
     <width>316</width>
     <height>500</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Shapes</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QTreeWidget" name="sceneGraph">
       <column>
        <property name="text">
         <string notr="true">Scene</string>
        </property>
       </column>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget_6" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="renameShape">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Rename</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="shapeRenderAsMesh">
       <property name="text">
        <string>Wireframe</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="MaterialWidget" name="shapeMaterial" native="true"/>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Scale</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QWidget" name="widget_4" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QSlider" name="scaleXYZ">
             <property name="maximum">
              <number>200</number>
             </property>
             <property name="singleStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="individualScaleControl">
             <property name="text">
              <string>Individual</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_5" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>X</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="scaleX">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Y</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="scaleY">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Z</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="scaleZ">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="scaleReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Position</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="translateX">
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="translateY">
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Z</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="translateZ">
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget_2" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="widget_3" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Rotation</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="rotationX">
          <property name="suffix">
           <string>°</string>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="rotationY">
          <property name="suffix">
           <string>°</string>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Z</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="rotationZ">
          <property name="suffix">
           <string>°</string>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="camera_dockWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Camera</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="CameraWidget" name="cameraWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="shaders_dockWidget">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Shaders</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="ShaderManagerWidget" name="shaderManager" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionCamera_Translation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/CameraDrag.png</normaloff>:/uGLWindow/Images/CameraDrag.png</iconset>
   </property>
   <property name="text">
    <string>Pan</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionCamera_Rotation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/CameraRotate.png</normaloff>:/uGLWindow/Images/CameraRotate.png</iconset>
   </property>
   <property name="text">
    <string>Pitch and yaw</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionCube">
   <property name="text">
    <string>Cube</string>
   </property>
   <property name="shortcut">
    <string>Shift+C</string>
   </property>
  </action>
  <action name="actionCameraDepth">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/CameraDepth.png</normaloff>:/uGLWindow/Images/CameraDepth.png</iconset>
   </property>
   <property name="text">
    <string>Dolly</string>
   </property>
   <property name="toolTip">
    <string>Camera depth</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionExport_as_PNG">
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/ExportAsPicture.png</normaloff>:/uGLWindow/Images/ExportAsPicture.png</iconset>
   </property>
   <property name="text">
    <string>PNG</string>
   </property>
   <property name="toolTip">
    <string>Export the scene as a PNG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionPNG">
   <property name="text">
    <string>PNG</string>
   </property>
  </action>
  <action name="actionImport_model">
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/ModelImport.png</normaloff>:/uGLWindow/Images/ModelImport.png</iconset>
   </property>
   <property name="text">
    <string>Import model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionCamera_Roll">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/CameraRoll.png</normaloff>:/uGLWindow/Images/CameraRoll.png</iconset>
   </property>
   <property name="text">
    <string>Roll</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R, Ctrl+O</string>
   </property>
  </action>
  <action name="actionCircle">
   <property name="text">
    <string>Circle</string>
   </property>
   <property name="shortcut">
    <string>Shift+C, Shift+I</string>
   </property>
  </action>
  <action name="actionSquare">
   <property name="text">
    <string>Square</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="actionPolygon">
   <property name="text">
    <string>Polygon</string>
   </property>
   <property name="shortcut">
    <string>Shift+P</string>
   </property>
  </action>
  <action name="actionHexagone">
   <property name="text">
    <string>Hexagone</string>
   </property>
   <property name="shortcut">
    <string>Shift+H</string>
   </property>
  </action>
  <action name="actionPyramid">
   <property name="text">
    <string>Pyramid</string>
   </property>
   <property name="shortcut">
    <string>Shift+P, Shift+Y</string>
   </property>
  </action>
  <action name="actionTriangle">
   <property name="text">
    <string>Triangle</string>
   </property>
   <property name="shortcut">
    <string>Shift+T</string>
   </property>
  </action>
  <action name="actionTexture">
   <property name="text">
    <string>Texture</string>
   </property>
  </action>
  <action name="actionBezier_curve">
   <property name="text">
    <string>Bezier curve</string>
   </property>
   <property name="shortcut">
    <string>Shift+C</string>
   </property>
  </action>
  <action name="actionBezier_surface">
   <property name="text">
    <string>Bezier surface</string>
   </property>
   <property name="shortcut">
    <string>Shift+S, Shift+U</string>
   </property>
  </action>
  <action name="actionGrass">
   <property name="text">
    <string>Grass</string>
   </property>
   <property name="shortcut">
    <string>Shift+G</string>
   </property>
  </action>
  <action name="actionWater">
   <property name="text">
    <string>Water</string>
   </property>
   <property name="shortcut">
    <string>Shift+W</string>
   </property>
  </action>
  <action name="actionModify_Skybox">
   <property name="icon">
    <iconset>
     <normaloff>Images/cubemap.png</normaloff>Images/cubemap.png</iconset>
   </property>
   <property name="text">
    <string>Modify Skybox</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S, Ctrl+K</string>
   </property>
  </action>
  <action name="action3D_rock">
   <property name="text">
    <string>3D rock</string>
   </property>
   <property name="shortcut">
    <string>Shift+R</string>
   </property>
  </action>
  <action name="actionVertigo">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="uglwindow.qrc">
     <normaloff>:/uGLWindow/Images/eye.png</normaloff>:/uGLWindow/Images/eye.png</iconset>
   </property>
   <property name="text">
    <string>Vertigo</string>
   </property>
   <property name="toolTip">
    <string>Vertigo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V, Ctrl+E</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>uGLSceneWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>uglscenewidget.h</header>
  </customwidget>
  <customwidget>
   <class>ShaderManagerWidget</class>
   <extends>QWidget</extends>
   <header>shadermanagerwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CameraWidget</class>
   <extends>QWidget</extends>
   <header location="global">camerawidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MaterialWidget</class>
   <extends>QWidget</extends>
   <header>materialwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="uglwindow.qrc"/>
 </resources>
 <connections/>
</ui>
